src/worker/mod.rs

```diff
@@ -337,10 +358,14 @@ impl<
         }))
     }
 
-    /// Run this worker on the given `queues` until an I/O error occurs (`Err` is returned), or
-    /// until the server tells the worker to disengage (`Ok` is returned).
+    /// Run this worker on the given `queues`.
     ///
-    /// The value in an `Ok` indicates the number of workers that may still be processing jobs.
+    /// Will run the worker until an I/O error occurs (`Err` is returned), or until the server tells the worker
+    /// to disengage (`Ok` is returned), or a signal from the user-space code has been received via a future
+    /// supplied to [`WorkerBuilder::with_graceful_shutdown`](`Ok` is returned).
+    ///
+    /// The value in an `Ok` indicates the number of workers that may still be processing jobs, but `0` can also
+    /// indicate the [graceful shutdown period](WorkerBuilder::graceful_shutdown_period) has been exceeded.
```

Comment:
Maybe we should introduce an `enum` here to tell the user which `Ok`-returning case they are in?

Reply:
Intrroduced `RunCeaseReason` and updated the docs and the tests